src/tests/test_bootstrap_in_place.py

```diff
@@ -237,11 +255,150 @@ def test_bootstrap_in_place(
+
+
+class TestBootstrapInPlaceWorker(TestBootstrapInPlace):
+    @pytest.fixture
+    def new_controller_configuration(self, request) -> BaseNodeConfig:
+        return TerraformConfig(
+            masters_count=1,
+            workers_count=1,
+            master_memory=16 * 1024,  # in megabytes
+            master_vcpu=16,
+            worker_memory=16 * 1024,  # in megabytes
+            worker_vcpu=16,
+            bootstrap_in_place=True,
+            running=False,
+        )
+
+    def render_worker_live_iso_ignition(self, install_device: str):
+        """
+        The worker live iso ignition file is embedded in the live ISO for the worker
+        and is responsible for:
+            - Copying the worker.ign file into the live filesystem
+            - Creating a one-shot systemd unit service which runs coreos-installer with the worker.ign
+            - Rebooting the node once the operating system has been written to disk
+
+        The worker then starts with the installed RHCOS+worker.ign and attempts to join the cluster
+
+        @param install_device The path of the disk to install RHCOS on (e.g. /dev/vda)
+        """
+        with open(os.path.join(RESOURCES_DIR, WORKER_LIVE_IGNITION_TEMPLATE), "r") as f:
+            live_iso_ignition_template_contents = f.read()
+
+        with open(os.path.join(RESOURCES_DIR, WORKER_INSTALL_SCRIPT), "rb") as f:
+            worker_install_script_contents = f.read()
+
+        with open(os.path.join(IBIP_DIR, "worker.ign"), "rb") as f:
+            worker_ignition_contents = f.read()
+
+        jinja2.filters.FILTERS["b64encode_utf8"] = lambda s: base64.b64encode(s).decode("utf-8")
+
+        return jinja2.Template(live_iso_ignition_template_contents).render(
+            ssh_public_key=os.environ["SSH_PUB_KEY"],
+            worker_ign_contents=worker_ignition_contents,
+            install_sh_contents=worker_install_script_contents,
+            install_device=install_device,
+        )
+
+    def worker_ready(self) -> bool:
+        try:
+            node_readiness_map = get_nodes_readiness(KUBE_CONFIG)
+            if not node_readiness_map:
+                log.debug("No nodes found or failed to list nodes. This is usually due to API downtime. Retrying")
+                return False
+
+            if not f"{CLUSTER_PREFIX}-master-0" in node_readiness_map:
+                log.warning("Couldn't find master in node status list, this should not happen")
+                return False
+
+            if not f"{CLUSTER_PREFIX}-worker-0" in node_readiness_map:
+                return False
+
+            return all(node_status for node_status in node_readiness_map.values())
+
+        except Exception as e:
+            print("got exception while validating operators: %s", e)
+            return False
+
+    @JunitTestCase()
+    def waiting_for_added_worker(self, controller: NodeController):
+        vm_ip = controller.master_ips[0][0]
```

Comment:
Unsure what is going on here, you get the master IP for collecting logs? why here?

Reply:
Fixed `log_collection` function to look like this:

```python
    @JunitTestCase()
    def log_collection(self, master_ip: Optional[str]):
        """
        Collects all sorts of logs about the installation process

        @param master_ip The IP address of the master node. Used to SSH into the node when doing installer gather.
                         When not given, installer gather log collection is skipped.
        """
        ...
```

Now the worker phase will just pass master_ip as None to skip useless installer gather